corehq/apps/sso/forms.py

```diff
@@ -618,6 +643,57 @@ def get_primary_fields(self):
             ),
         ]
 
+    def get_remote_user_management_fields(self):
+        def mask_api(s):
+            if len(s) <= 3:
+                return s
+            else:
+                return s[:3] + '*' * (len(s) - 3)
+        masked_api = mask_api(self.idp.api_secret)
```

Comment:
If I save a form, the next time I view the form, the API key should now be obfuscated. If I'm following the code correctly, if I try to edit this form but do not edit the API key (perhaps I just want to edit the name, for example), it looks like I will send back an obfuscated API key, and that will overwrite whatever was previously saved.

I'd imagine the flow we want to support is to allow unrelated changes to be made that don't overwrite the valid API key saved in our database?

In the event that the user does try to modify the API key, it might also make sense to include validation logic that checks for the obfuscation strings (i.e. warning on finding '***')

Reply:
Hi Matt, you wend send back an obfuscated API key even if you didn't edit the key.

Reply:
@biyeun we can resolve this comment, right? It seems the API key is no longer revealed